app/controllers/clients/files_controller.rb

```diff
@@ -107,7 +107,10 @@ def update
         attrs[:effective_date] = attrs[:consent_form_signed_on]
         attrs[:consent_form_confirmed] = true if GrdaWarehouse::Config.get(:auto_confirm_consent)
       end
-      @file.update(attrs)
+      @file.assign_attributes(attrs)
+      revoked_by_id = current_user.id if @file.consent_revoked_at.present?
+      @file.consent_revoked_by_user_id = revoked_by_id if @file.consent_revoked_at_changed?
```

Comment:
```suggestion
      if @file.consent_revoked_at_changed?
        @file.consent_revoked_by_user_id = if @file.consent_revoked_at.present?
          current_user.id
         else
           nil
         end
       end
```

Reply:
The above can go in a method on `ClientFile` so you can call `@file.set_revocation_info(current_user)`